grpc/runtime/src/main/scala/io/buoyant/grpc/runtime/ServerDispatcher.scala

```diff
@@ -148,7 +149,13 @@ class ServerDispatcher(services: Seq[ServerDispatcher.Service])
       }
     }.toMap
 
-  override def apply(req: h2.Request): Future[h2.Response] =
+  override def apply(req: h2.Request): Future[h2.Response] = {
+    // According to gRPC spec, server must return 415 if content-type is not set to
+    // application/grpc to allow for graceful error handling when plain H2 client
+    // makes request to a gRPC endpoint.
+    // This, however, would be a breaking change on linkerd side.
+    // Postponing content-type validation for now.
+    // https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md
```

Comment:
Maybe add an issue as a reminder that this needs to be implemented for the next major release

Reply:
:+1: to filing a github issue and linking to it in this comment